fs/dafsserver/dafsserver.cpp

```diff
@@ -1308,6 +1321,34 @@ class CRemoteDiskBaseActivity : public CSimpleInterfaceOf<IRemoteReadActivity>,
             ForEach(*filterIter)
                 filter.addFilter(*record, filterIter->query().queryProp(nullptr));
         }
+
+        if (config.hasProp("recordSamplingRate"))
+        {
+            // std::bernoulli_distribution may support smaller sampling rates, depending on implementation,
+            // but we limit it to 1e-12 to avoid issues with very small probabilities
+            constexpr double minSamplingRate = 1e-12;
+            constexpr double maxSamplingRate = 1.0;
+
+            double recordSamplingRate = config.getPropReal("recordSamplingRate", maxSamplingRate);
+            if (recordSamplingRate <= minSamplingRate || recordSamplingRate > maxSamplingRate)
+                throw createDafsException(DAFSERR_cmdstream_protocol_failure, "CRemoteDiskBaseActivity: recordSamplingRate must be between (1e-12, 1.0)");
+
+            if (recordSamplingRate < maxSamplingRate)
+            {
+                applySampling = true;
+                recordSamplingDistribution = std::bernoulli_distribution(recordSamplingRate);
+            }
+
+            __int64 recordSamplingSeed = config.getPropInt64("recordSamplingSeed", -1);
```

Comment:
minor: This could move into the previous if - avoiding minor work if sampling rate is set to 1.0